http-server/src/test/java/io/airlift/http/server/TestHttpServerProvider.java

```diff
@@ -392,18 +395,33 @@ public void testShowStackTraceDisabled()
     public void testStop()
             throws Exception
     {
-        createAndStartServer();
-
+        DummyServlet servlet = new DummyServlet();
+        createAndStartServer(servlet);
         try (HttpClient client = new JettyHttpClient()) {
             URI uri = URI.create(httpServerInfo.getHttpUri().toASCIIString() + "/?sleep=50000");
             Request request = prepareGet().setUri(uri).build();
-            HttpResponseFuture<?> future = client.executeAsync(request, createStatusResponseHandler());
+            HttpResponseFuture<StatusResponse> future = client.executeAsync(request, createStatusResponseHandler());
 
+            // wait until the servlet starts processing the request
+            while (servlet.getActiveRequests() == 0) {
+                Thread.sleep(100);
```

Comment:
You could use a latch here also, but I'm not sure it is worth the hassle.

Reply:
Use a `CountDownLatch` here